---
title: Supplementary Materials \newline
  Growth-mediated dilution controls the response of gene regulatory networks
author:
  - name: Thomas Julou
    # email: thomas.julou@normalesup.org
    # corresponding: TRUE
    # affiliation: 
    #   - BZ
    #   - SIB
  - name: Athos Fiori
    # affiliation: BZ
  - name: Diana Blank
    # affiliation: BZ
  - name: Erik van Nimwegen
    # email: erik.vannimwegen@unibas.ch
    # corresponding: TRUE
    # affiliation: 
    #   - BZ
    #   - SIB
address:
  - code: BZ
    address: Biozentrum, University of Basel, Klingelbergstrasse 50/70, 4056 Basel, Switzerland.
  - code: SIB
    address: Swiss Institute of Bioinformatics.

# author_summary: |
#   *to be written*

# output: rticles::plos_article
output:
  bookdown::pdf_book:
    base_format: rticles::plos_article
csl: plos.csl
bibliography: MoM_lacDilution_SM.bib

header-includes:
  \usepackage{float} \floatplacement{figure}{p}
  \setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}
  \usepackage{xr} \externaldocument{MoM_lacDilution_ms}
  \usepackage{booktabs}
  # \usepackage{tabu}
  # \usepackage{varwidth}
  # \usepackage{longtable} 
---


```{r include=FALSE}
knitr::opts_chunk$set(echo = FALSE, out.width='4.75in')
```


# Bacterial strains and media
The main strain used in this study is ASC662 (MG1655 lacZ-GFPmut2) [@kiviet2014stochasticity]. 
We confirmed by sequencing that this strain has no mutation in the *lac* operon apart from the lacZ-GFP translational fusion, and identified the GFP sequence as a GFPmut2 lacking the monomeric substitution A206K; fortunately this doesn’t affect its spectrum nor its maturation time [@balleza2018systematic].
In order to increase LacI activity in the repressed state, plasmid p_lacI_SC101 was transformed into ASC662. It was obtained by restriction-ligation cloning of MG1655's native lacI promoter and gene (amplified by PCR between positions 366,428 and 367,600) into a pSC101 vector (pUA66; using XhoI and XbaI, hence removing the plasmid's promoter and GFP loci). In order to make cloning easier, we used a high copy number derivative of pUA66 where the GFP locus is replaced with pUC19 origin (produced by restriction-ligation cloning with BamHI and XbaI).

All experiments were done using M9 minimal media supplemented with 2 mM MgSO4, 0.1 mM CaCl2, and sugars as indicated (typically 0.2% for glucose or lactose and 0.4% for glycerol). TMG and IPTG were diluted from frozen stocks in water (at 0.1 M and 1 M respectively).

All experiments were carried out at 37ºC. xxx melibiose transporter.

# Microfluidic device fabrication
The Dual Input Mother Machine (DIMM) microfluidic design used in this study has been described elsewhere [@kaiser2018monitoring] and is freely available online [@dimm]: in brief it combines comb-like structures enabling to grow bacteria in steady state conditions over long times with a special type of junction allowing fast and accurate mixing of two input media. Several microfluidics masters were produced using soft lithography technics by micro-resist Gmbh; one master with regular growth channels of suitable size (0.7 µm width × 0.9 µm height) was used for all experiments.

For each experiment, a new chip was produced by pouring PDMS (Sylgard 184 with 1:9w/w ratio of curing agent) on the master and baking it for 4 h or more at 80ºC. After cutting the chip and punching inlets, the chip was bonded to a #1.5 glass coverslip as follows: the coverlsip was sonicated 3 to 5 min in acetone, rinsed in water then isopropanol; the chip cleaned from dust using MagicTape, rinsed in isopropanol then water; surfaces were activated with air plasma (40 sec at 1500 µm of Hg) before being put in contact; the assembled chip was cooked 1 h or more at 80ºC.

Before running the experiment, the chip was primed with passivation buffer (2.5 mg/mL salmon sperm DNA, 7.5 mg/mL bovine serum albumin) for the mother machine part and with water for the part with overflow channels, and incubated 1 h at 37ºC.

# Experiment setup and conditions
Bacteria were streaked onto LB agar plates from frozen glycerol stocks thawed from −80ºC. Overnight precultures were grown from single colonies in M9 minimal medium supplemented with the same sugar that the cells were to experience in the initial condition of the experiment. The next day, cells were diluted 100-fold into fresh medium with the same sugar and harvested after 4–6 h. 

The experimental apparatus was initialized, pre- warmed and equilibrated. Flow control was achieved using a syringe pump for early experiments and using a pressure controller later on. A total flow of ≈3 µL/min was used in all cases (corresponding to a pressure of ≈2000 mbar on both inlet). Polystyrene beads (Polybead 1 μm) were added to one of the two media to allow controlling the flow at the mixing junction.
<!-- see table -->

The primed microfluidic chip was mounted, connected to media supply and flushed with running media for 30 min or more to rinse passivation buffer. The grown cell culture was centrifuged at 4000×g for 5 min, and the pellet resuspended in a few μl supernatant and injected into the device using a syringe. Since flow was running, the pressure had to be continuously adjusted to make sure the cells stopped flowing in the main channel and could enter the growth channels (typically 10 to 40 min).

After loading, bacteria were incubated during 2 h before starting image acquisition. Every 3 min (6 min for experiments in glycerol), phase contrast and fluorescence images were acquired for several positions in parallel (typically 6), including the dial-a-wave junction with short phase contrast exposure (10 ms) so as to control the flow ratio between the two inputs. 

<!-- # Supplementary tables -->

```{r expts-list}
#  xxx add comments + flow control
# remove lac_iptg
mytables[['expts_list']] %>% 
  ungroup() %>% select(-condition) %>% rename(condition=label) %>% 
  mutate(condition=str_replace(condition, '>', ' to ')) %>% 
  (function(.df)
    knitr::kable(.df, "latex", booktabs=TRUE, 
                 col.names=c('condition', 'date', '# growth channels', '# full cell cycles', '# observations',
                             '# cells at switch', '# estimated lags', '# arrested cells at switch'),
                 caption='List of experiments used in this study with summary statistics. Experiments that have been discarded from further analysis are greyed out.') %>% 
     # kableExtra::kable_styling(full_width=TRUE) %>% 
     kableExtra::kable_styling(latex_options = c("striped", "scale_down")) %>%
     kableExtra::column_spec(3:8, width = "3.5em") %>% 
     kableExtra::row_spec(which(.df$date %in% discarded_dates), italic=T, color="gray") %>% 
     identity()
  ) 

```


# Microscopy and image analysis
An inverted Nikon Ti-E microscope, equipped with a motorized xy-stage and enclosed in a temperature incubator (TheCube, Life Imaging Systems), was used to perform all experiments. The sample was fixed on the stage using metal clamps and focus was maintained using hardware autofocus (Perfect Focus System, Nikon). Images were recorded using a CFI Plan Apochromat Lambda DM ×100 objective (NA 1.45, WD 0.13 mm) and a CMOS camera (Hamamatsu Orca-Flash 4.0). The setup was controlled using µManager [@edelstein2010computer] and timelapse movies were recorded with its Multi-Dimensional Acquisition engine (customized using runnables). Phase contrast images were acquired using 100 ms exposure (CoolLED pE-100, full power). Images of GFP fluorescence (ex 475/35 nm; em 525/50 nm; bs 495 nm) were acquired using 2 s exposure (Lumencor SpectraX, Cyan LED at 17% with ND4). 

Image analysis was performed using MoMA [@kaiser2018monitoring] as described in its documentation [@moma]. Raw image datasets were transfered to a centralised storage and preprocessed in batch. For each experiments, 30 growth channels were picked randomly (after discarding channels with structural defects or no cells growing at the first switch of condition) and curated manually in MoMA. MoMA's default post-processing was used in order to refine the measurements of cell length and total fluorescence. Fluorescence arbitrary units were converted to number of GFP molecules using the procedure and conversion factors described previously [@kaiser2018monitoring].

# Estimation of *lac* induction lags
The time until the *lac* operon is induced in a given cell after a switch to lactose is estimated from the time series of LacZ-GFP level as follows: we compute the pre-induction level computed as the mean value in the 9 min following the switch (for which we checked visually that no cell had induced yet) and measure the delay after the switch until the cell has increased its level by 200 molecules (which is a conservative threshold since the fluctuations of background level have a standard deviation of ≈30); hence we need to observe a cell for at least 4 time points after the switch to be able to successfully detect an induction. In order to successfully estimate a lag for as many cells as possible, if a cell divided before inducing, the total GFP of its two daughters was summed and their cell traces considered until the induction threshold was passed, or one of the two daughters was lost (because it left the channel). We note that this *lac* induction lag will include the maturation time of the LacZ-GFP fusion proteins (half time = 5.6 ± 0.4 min @balleza2018systematic).

xxx proba not accouting for exiting cells

In order to analyse transcriptional memory, we would like to know the number of LacZ-GFP molecules in each bacteria at the onset of the switch to lactose. However, the combination of limited sensitivity (on the order of dozens of GFP molecules per cell) and spontaneous autofluorescence (of similar magnitude), prevents us from accurately measuring levels lower than 100 molecules; in addition, repeated illumination induces bleaching so that not all GFP molecules are fluorescent (which has been quantified to ≈20% per cell cycle; [@kaiser2018monitoring]). However, at the second switch, the number of LacZ-GFP inherited from the ancestors' induction during the first exposure can be computed as follows: the lineage is traced backed is time and both $L_m$, the maximum LacZ-GFP level during the first exposure, as well as $n_d$, the number of divisions since this time, are recorded; the number of inherited molecules follows as $L_m / 2^{n_d}$. Note that this formula assumes no degradation (which is reasonable since the longest delay is 24 h) and even partitioning at every division, and that this constitutes a lower bound for the number of LacZ-GFP molecules in the cell since stochastic expression may have happened during the delay period.

# Estimation of growth lags


# Quality control and data filtering
<!-- visual: xxx try to show interquartile instead of 95pi -->

Given the design of the DIMM microfluidic chip, only one strain and one condition can be used in each experiment. Hence we characterized cells growing in initial conditions in order to make sure that they were in similar physiological states (between experiments) when the switch of condition occured; otherwise differences in responses could simply come from differences in initial state. For each full cell cycle observed before the first switch, we fit a linear model to log(length) and time and compute the growth rate, the Pearson correlation coefficient and the predicted initial cell length. While distribution of correlations coefficient are difficult to compare between days, the growth rate and initial cell length display as expected a clear correlation for each condition (Fig. \@ref(fig:gr-length-medians)). This supports that changes in growth rate truly indicate modulation of physiological states (rather than e.g. merely steming from mechanical constraints imposed by the microfluidic chip). Noticeably the variability between experiments for one condition is not small, on the same magnitude as the variability between cells in one experiment, and no clear clusters can be distinguished which makes it difficult to define criteria for initial conditions to be comparable. Nonetheless, 4 experiments appear as obvious outliers with slow growing cells and were discarded as such (20151218, 20161021, 20170901, 20180313); in addition, we discarded 2 more experiments with slow growing cells (20160526, 20170108). Furthermore we examined the reproducibility of *lac* induction lags distributions between replicates or closely-related conditions and discarded 2 additional experiments based on this criteria; although one had a fraction of short lags similar to its replicates (20180123), the other was qualitatively different to our expectations (20180615) and no reason could be found for this discrepancy (leaving the door opened to an undocumented protocol mistake). It should be noted that the dataset used here has been acquired over almost 3 years which is very unusual for this type of studies and imposes new challenge regarding the comparison of experiments.

Since our study focus on the response of *growing* cells to changes of environmental conditions, we discarded a small fraction of bacteria that were not growing before the switch and were most of the time unable to induce at all. The growth rate before the switch was measured by fitting a linear model to log(length) and time; in case the cell was born less than 15 min before the switch, the growth rate of its parent was used. Based on the relationship of *lac* induction and growth rate before the switch, it was possible to set a growth rate threshold that doesn't truncate the main mode of the distribution (Fig. \@ref(fig:gr-before-filtering)). The number of bacteria discarded in each experiment is reported in Table \@ref(tab:expts-list) and corresponds to a small fraction of cells present at the switch (below 3% in 0.2% glucose and below 10% in 0.4% glycerol).

Further analysis was run using these two filtering criteria.

# Direct measurement of LacZ-GFP xxx

use 50x stronger illumination for GFP.
compare intensity histo 

Fig \@ref(fig:high-illum-sensitivity)


# Decreasing LacI activity with IPTG

binding / unbinding time U. Alon


# Decoupling induction and growth using lactulose and TMG

cells with high [GFP] but slow growth may come from errors in gr estimation (length outliers) or delay between LacZ expression and growth (GFP maturation + metabolism cf Kiviet)




<!-- # Supplementary figures -->

(ref:gr-length-medians-caption) Physiological state in initial conditions. For each experiment, the medians (± 95% posterior intervals) are plotted for growth rate and cell length at birth. The 2 clusters seen in glycerol correspond to different illumination conditions.

```{r gr-length-medians, fig.height=3.5*14/9, fig.cap="(ref:gr-length-medians-caption)"}
myplots[['gr_length_medians']]
```


<!-- (ref:gr-before-filtering-caption) Growth arrest before the switch prevents *lac* induction. For each cell, the *lac* induction lag is plotted as a function of the growth rate before the switch (top). A growth rate threshold (dotted line) is chosen so as not to truncate the main mode of the marginal distributions of growth rates (bottom). xxx *this is not necessary for these datasets : skip it or describe it as a criteria for robust analysis including other datasets*. -->

<!-- ```{r gr-before-filtering, fig.height=3.5*14/9, fig.cap="(ref:gr-before-filtering-caption)"} -->
<!-- plot_grid( -->
<!--   myplots[['lags_gr_before']](.size_n=FALSE) + -->
<!--     theme(legend.position=c(1,1), legend.justification=c(1,1), -->
<!--           axis.title.x=element_blank()) + -->
<!--     NULL, -->
<!--   myplots[['gr_hist_before']] + -->
<!--     theme(legend.position='none') + -->
<!--     NULL, -->
<!--   ncol=1, rel_heights=c(1.5, 1), align='v' -->
<!-- ) -->
<!-- ``` -->


(ref:naive-lags-per-expt-caption) Distribution of induction lags for the *lac* operon in naive cells, stratified per day. The first panel shows all days overlaid which highlights that despite the varibility between replicates, the bimodality of *lac* induction lags distributions is a robust feature.

```{r naive-lags-per-expt, fig.height=6*14/9, fig.cap="(ref:naive-lags-per-expt-caption)"}
myplots[['naive_lags_per_expt']]
```


(ref:lags-hist-ramp-caption) Distribution of induction lags for the *lac* operon in naive cells exposed to a gradual transition from 0.2% glucose to 0.2% lactose over 40 min (2 independent replicates). It is remarkable that this distribution is still bimodal since it indicates that all cells detect a very similar (although unknown) critical concentration of glucose and/or lactose. For each replicate the distribution of lags is offset by the delay that best overlays it to the control one (since the critical concentration at which the switch is detected is unknown). xxx explain more + xxx try to detect growth arrest

```{r lags-hist-ramp, fig.cap="(ref:lags-hist-ramp-caption)"}
myplots[['lags_hist_ramp']]
```


(ref:lag-memory-caption) *lac* induction lag at the second exposure to lactose as a function of the ancestor's lag at the first exposure. Dotted lines show $y=x$ as an eyeguide.

```{r lag-memory, fig.height=4*14/9, fig.cap="(ref:lag-memory-caption)"}
myplots[['lag_memory']]
```


(ref:memory-cdfs-facets-caption) Reverse cumulative distributions of *lac* induction lags at the first and second exposure to lactose. Each line correspond to an independent experiment; colours indicate the delay between the two lactose exposures. At the first switch, all experiments correspond to the same condition (the variability corresponds to measurement noise and biological variability between experiments; data shown as histograms on Fig \@ref(fig:naive-lags-per-expt)) and a negative control where cells are first exposed to lactose after 26 h in the microfluidics chip is shown in light gray.

```{r memory-cdfs-facets, fig.height=4*14/9, fig.cap="(ref:memory-cdfs-facets-caption)"}
myplots[['memory_cdfs_facets']]
```


(ref:memory-elapsed-divs-caption) Histogram of number of elapsed division stratified by delay between the two lactose exposures. The even spacnig between distributions is expected since delays increases evenly between conditions.

```{r memory-elapsed-divs, fig.cap="(ref:memory-elapsed-divs-caption)"}
myplots[['memory_elapsed_divs']]
```


(ref:lags-inherited-gfp-caption) *lac* induction lag at the second switch as a function of the number of inherited LacZ-GFP. Each cell with successful lag estimation is shown as a dot; the gray line corresponds the prediction (with standard error) using local regression (loess with neighbourhood including 20% of the points and tricubic weighting).

```{r lags-inherited-gfp, fig.height=3*14/9, fig.cap="(ref:lags-inherited-gfp-caption)"}
myplots[['lags_inherited_gfp']]
```


(ref:high-illum-sensitivity-caption) xxx.

```{r high-illum-sensitivity, fig.cap="(ref:high-illum-sensitivity-caption)"}
(myplots[['high-illum-sensitivity']] <-
bind_rows(
  read.csv("../data/20180403_glu_lac_hiIllum_1_pos2_t215_MG1655_hist.csv") %>% 
    with(rep.int(value, count)) %>% tibble(strain='MG1655', int=.),
  read.csv("../data/20180403_glu_lac_hiIllum_1_pos2_t215_ASC662_hist.csv") %>% 
    with(rep.int(value, count)) %>% tibble(strain='ASC662', int=.) 
  ) %>% 
  ggplot(aes(int, ..density.., col=strain)) +
  geom_freqpoly(binwidth=50) +
  scale_colour_discrete(breaks=c('MG1655', 'ASC662'), labels=c('MG1655 (70 cells)', 'ASC662 (93 cells)')) +
  labs(x='pixel intensity (AU)') +
  NULL)

```



<!-- (ref:basal-perturb-with-caption) Distribution of *lac* induction lags with increased (green) or reduced (blue) LacI activity corresponding to lower or higher expression respectively in the repressed state. Each contour corresponds to an independant replicate (with ≈150 cells). These experiments are similar to Fig. 2A but with LacI titrating before and during the exposure to lactose (instead of only during the exposure). -->

<!-- ```{r basal-perturb-with, fig.cap="(ref:basal-perturb-with-caption)"} -->
<!-- myplots[['basal_perturb_with']] -->
<!-- ``` -->


(ref:basal-perturb-gfp-caption) LacZ-GFP expression level before exposure to lactose when LacI activity is reduced by IPTG titration. Although this treatment increases the fraction of fast-switching cells, no measureable change of LacZ-GFP levels can be detected which supports that the increase of basal expression is less than 100 molecules. In comparison, bacteria carry 3000-6000 LacZ-GFP molecules at full induction (Fig. 1A).

```{r basal-perturb-gfp, fig.cap="(ref:basal-perturb-gfp-caption)"}
myplots[['basal_perturb_gfp']]
```


<!-- (ref:lags-hist-glyc-glcLac-caption) Histograms of *lac* induction lags for initial conditions where the basal expression is expected to be higher: 0.4% glycerol and 0.2% glucose + 0.2% lactose (2 independant replicates per condition). The corresponding fractions of short lags are shown in Fig. 2B. -->

<!-- ```{r lags-hist-glyc-glcLac, fig.cap="(ref:lags-hist-glyc-glcLac-caption)"} -->
<!-- myplots[['lags_hist_glyc_glcLac']] -->
<!-- ``` -->








# References
